## Import Statements

In [178]:
import gensim.downloader as api
import numpy as np
import re
import csv
import pandas as pd
import pprint
import string
import nltk
import sys
import spacy
import os
import pickle
    
from IPython.display import HTML
from nltk.corpus import wordnet 
from sklearn.manifold import TSNE
from gensim.models import Word2Vec
from gensim.models import KeyedVectors

from nltk.sentiment.vader import SentimentIntensityAnalyzer
from nltk.tokenize import word_tokenize

pathToDatasets = '../datasets/'
pathToDataScripts = '../datasets/scripts/'
filePath = '../datasets/GoogleNews-vectors-negative300.bin'
# modelBeingUsed = "glove-wiki-gigaword-300"
# modelBeingUsed = "glove-wiki-gigaword-100"
# modelBeingUsed = "glove-twitter-50"
# modelBeingUsed = "glove-twitter-100"
modelBeingUsed = "glove-twitter-200"

sys.path.insert(0, pathToDataScripts)
from cleanDataset import tokenize_words, dataClean


from IPython.core.display import display, HTML
display(HTML("<style>.container { width:85% !important; }</style>"))

## Downloading binaries and models


In [179]:
print("Should I reload the model?")
tstString = input()
if("no" in tstString.lower() or "n" in tstString.lower()):
    print(" didnt reload model! ")
else:
    print("loading {0}!".format(modelBeingUsed));
    fileName = "{}.pickle".format(modelBeingUsed)
    if(os.path.exists(pathToDatasets+fileName)):
        print("loading via pickle!")
        pickle_in = open(pathToDatasets+fileName, "rb")
        word_vectors = pickle.load(pickle_in);
    else:
        print("Pickle didn't exist, therefore loading model!")
        word_vectors = api.load("{0}".format(modelBeingUsed))
        print("-- Saving to pickle file for next time! --")
        pickle_out = open(pathToDatasets+fileName,"wb")
        pickle.dump(word_vectors, pickle_out)
        pickle_out.close()
    nltk.download('vader_lexicon')
    nltk.download('punkt')
    nltk.download('averaged_perceptron_tagger')
    nltk.download('wordnet')
    print("Model Loaded!")

Should I reload the model?
no
 didnt reload model! 


## Global Variables and Global Objects

In [180]:
senty = SentimentIntensityAnalyzer()
vocabulary = word_vectors.vocab;

nlp = spacy.load('en')


'''
Good sets -> {100+20}
'''
NUMBER_OF_ALTERNATIVES = 7 
TWEET_START = 150
NUM_OF_TWEETS = 50


# VERBOSE_PRINTING = True
VERBOSE_PRINTING = False

# USE_SPACY = False
USE_SPACY = True

COLOR_PRINTING = True
#COLOR_PRINTING = False

PRINT_NEUTRAL = True
# PRINT_NEUTRAL = False

PRINT_ALL_STRINGS = True
# PRINT_ALL_STRINGS = False

SHOW_ALTS = 50

punctuation = r"\"#$%&'+-/;<=>?@[\]*^_`{|}~"
LIST_OF_NEGATIONS = ["not"]

## Class for Sentences


In [181]:
from SentenceClass import *

## Utility Code

In [182]:
def printStrings(sentenceObj):
    
    numberOfPrints = 0
    newStrings = generateHTMLObjectsFromSentence(sentenceObj)
    mainSentiment = sentenceObj.ogSentiment;
    listOfSentencesWithSentiments = []
    bestSentiment = -sys.maxsize - 1
    worstSentiment = sys.maxsize
    
    bestSentimentString = "";
    worstSentimentString = "";
    
    for ind, tSentence in enumerate(newStrings):
            alteredTweet = tSentence.getSentence()
            htmlText = tSentence.getHTML()
            
            if("not" in alteredTweet): continue
            
            sentimentOfNewString = senty.polarity_scores(alteredTweet)['compound']
            newObj = SentenceWithSentiment(alteredTweet, sentimentOfNewString, htmlText)
            listOfSentencesWithSentiments.append(newObj)
            
            if(sentimentOfNewString >= bestSentiment):
                bestSentiment = sentimentOfNewString
                bestSentimentString = htmlText;
            elif(sentimentOfNewString < worstSentiment):
                worstSentiment = sentimentOfNewString
                worstSentimentString = htmlText;
            
            
            numberOfPrints+=1;
            if(numberOfPrints > SHOW_ALTS or PRINT_ALL_STRINGS == False): continue
            if(sentimentOfNewString == mainSentiment or sentimentOfNewString == 0.0):
                if(PRINT_ALL_STRINGS and numberOfPrints <= SHOW_ALTS): displayText("{0}: {1}".format(htmlText,sentimentOfNewString),'black')
            elif(sentimentOfNewString > mainSentiment):
                if(PRINT_ALL_STRINGS and numberOfPrints <= SHOW_ALTS): displayText("{0}: {1}".format(htmlText,sentimentOfNewString),'green')
            elif(sentimentOfNewString < mainSentiment and sentimentOfNewString != 0.0):
                if(PRINT_ALL_STRINGS and numberOfPrints <= SHOW_ALTS): displayText("{0}: {1}".format(htmlText,sentimentOfNewString),'red')
                
    if(numberOfPrints > SHOW_ALTS): print("--- More options (total: {0}) possible, but not printed ---".format(numberOfPrints));
    displayText("Actual Sentence: {0} :{1}".format(sentenceObj.ogSentence, sentenceObj.ogSentiment))
    if (worstSentimentString != ""): displayText("Worst Sentence: {0} : {1}".format(worstSentimentString, worstSentiment), color='red')
    if (bestSentimentString != ""): displayText("Best Sentence: {0} : {1}".format(bestSentimentString, bestSentiment), color='green') 
    
    
    sentenceObj.addFinalSentences(listOfSentencesWithSentiments)
    return listOfSentencesWithSentiments, sentenceObj;

def cstr(s, color='black', italics=False):
    if(COLOR_PRINTING):
        if(italics):
            return cstr("<i>{0}</i>".format(s), color);
        return "<text style=color:{}>{}</text>".format(color, s)
    else:
        return "{}".format(s)

def displayText(text, color='black'):
    if(COLOR_PRINTING):
        display(HTML(cstr(text, color)));
        return
    print("{}".format(text));
    
    
def cleanAndTokenizeText(text):
    text = text.lower();
    newString = ""
    for char in text:
        if char not in punctuation:
            newString += char
    text = word_tokenize(newString)
    return text;

def getPOSTags(tweet):
    if(USE_SPACY == False):
        tags = nltk.pos_tag(tweet)
        return tags;    
    tweet = ' '.join(tweet)
    doc = nlp(tweet)
    tags = [(token.text, token.tag_) for token in doc] # since the format expected is [text,tag]
    return tags

def getAntonymsAndSynonymsOfWords(word):
    if(word not in vocabulary):
        return [], []
    setOfAntonyms = set()
    setOfSynonyms = set()
    for syn in wordnet.synsets(word):
        for l in syn.lemmas():
            setOfSynonyms.add(l.name().lower())
            anton = l.antonyms()
            if(anton!=[]):
                setOfAntonyms.add(anton[0].name().lower())
    if(len(setOfAntonyms) == 0):
        if(VERBOSE_PRINTING): print("No antonyms found for word {0}".format(word))
    if(len(setOfSynonyms) == 0):
        if(VERBOSE_PRINTING): print("No synonyms found for word {0}".format(word))
            
    return list(setOfAntonyms), list(setOfSynonyms)


def returnReplacementsForWord(word):
    
    if(word not in vocabulary):
        print(" --- {0} not in vocabulary ---".format(word))
        return []
    possibleReplacements = [word[0] for word in word_vectors.most_similar(word,topn=NUMBER_OF_ALTERNATIVES)]
    
    if(possibleReplacements == []):
        print(" --- No replacements for word {0} ---".format(word))
    antonyms,synonyms = getAntonymsAndSynonymsOfWords(word)
    
    if(antonyms != []):
        possibleReplacements.extend(antonyms)
        if(VERBOSE_PRINTING): print("Some antonyms for word {0} are {1}".format(word, antonyms[:3]))
    if(synonyms != []):
        possibleReplacements.extend(synonyms)
        if(VERBOSE_PRINTING): print("Some synonyms for word {0} are {1}".format(word, synonyms[:3]))
    return possibleReplacements
    
def posApprovedReplacements(alternativeWords, userTokens, indexOfToken):
    if(alternativeWords == []):
        return []
    tempTokens = userTokens[:]
    POSTokens = getPOSTags(tempTokens)
    validWords = []
    
    mainTag = POSTokens[indexOfToken][1]
    mainWord = userTokens[indexOfToken]
    
    for ind,word in enumerate(alternativeWords):
        if("_" in word): continue
        tempTokens[indexOfToken] = word
        posTags = getPOSTags(tempTokens)
        newTag = (posTags[indexOfToken])[1]
        if(str(newTag) == str(mainTag)):
#             if(VERBOSE_PRINTING): print("Word {0}[{1}] matched with {2}[{3}]".format(mainWord, mainTag, word,newTag))
            validWords.append(word)
    if(validWords == [] and VERBOSE_PRINTING):
        print("No POS words found for word {} with tag {}".format(mainWord, mainTag));
    return validWords
    
def addBoostersToSentence(sentenceObj):
    finalSentenceObjs = sentenceObj.getFinalSentences();
    newObjs = []
    counterOfAdditions = 0
    '''
    
    
    '''
    return
    
def generateHTMLObjectsFromSentence(sentenceObj):
    
    allSentences = sentenceObj.getFinalSentences()
    indexToAlts = sentenceObj.indexToSetOfWords;
    indexToChange = list(indexToAlts.keys());
    
    numberOfNegationsRemoved = 0;
    listOfSentenceObjs = []
    for t_sentenceObj in allSentences:
        sentence = t_sentenceObj.getSentence()
        sentence = t_sentenceObj.getSentence()
        copySentence = cleanAndTokenizeText(sentence)
        subIndex = 0
        for index in sorted(indexToChange):
            if(index  len(copySentence)):  
                copySentence[len(copySentence) - index] = cstr("[{0}]".format(copySentence[len(copySentence) - index]), "blue", italics=True);
                continue
            else:
                if("IGNORE_FLAG" in indexToAlts[index]):
                    if(copySentence[index] in LIST_OF_NEGATIONS):
                        continue; ## Word that _should_ be removed is present.
                    else:
                        subIndex +=1
                        numberOfNegationsRemoved +=1;
                        continue;
            copySentence[index-subIndex] = cstr("[{0}]".format(copySentence[index-subIndex]), "blue", italics=True);
        listOfSentenceObjs.append(SentenceWithHTML(sentence, ' '.join(copySentence)));
    
    if(VERBOSE_PRINTING): print("Number of negations removed : {0}".format(numberOfNegationsRemoved))
    return listOfSentenceObjs
                

## Sentence Chunking and Appending

In [183]:
def helper_combine(mainList, myList):
    '''
    helper function for CombineSentenceChunks
    '''
    newList = []
    for val in myList:
        for mainVal in mainList:
#             if(VERBOSE_PRINTING): print("Combining {0} with {1}".format(' '.join(val), ' '.join(mainVal)));
            newList.append(val + mainVal);
    return newList;

def combineSentenceChunks(wholeSentence, dictOfChunks):
    '''
        Uses the helper_combine function to generate all possible combinatios and permuations of the chunks
        and any alternatives.
        
        The logic is to use the end of the sentence, apply each possible chunk from the previous key's chunks
        to every possible chunk of this key's.
        
        The helper function is used to allow us to reuse the list of alreadyGeneratedChunks and constantly
        append to them.
        
        To understand the logic better, take a look at this gist:
        https://gist.github.com/sunnyMiglani/cf85407a9e6928237b1436cc2bc95fa4
        
    '''
    reversedKeys = sorted(dictOfChunks.keys(), reverse=True)
    completeSentences = [];
    mainArr = dictOfChunks[reversedKeys[0]]
    for ind in range(1, len(reversedKeys)):
        key = reversedKeys[ind];
        mainArr = helper_combine(mainArr, dictOfChunks[key]);
        
    return mainArr;
        
def generateSentenceChunks(wholeSentence, keyToChange, nextKey, listOfMyAlternatives):
    '''
        Generates sentence chunks by iterating through the list of alternatives
        Chunking the sentence to start from current key to next key.
        This means that the sentence always goes from key 'x' to key 'y'
        
        Example:
        "I really <hate> hot chocolate, but I <prefer> hot coffee"
        Calling generateSentenceChunks will create an example sentence:
            - "<altWordForHate> hot chocolate , but I "
        
        Remember to append the first stretch of the string to the first key's chunk for proper use!
    '''
    
    newList = list(listOfMyAlternatives)
    newList.append(wholeSentence[keyToChange]);
    generatedSentences = []
    
        
    for myAlt in newList:
        if("IGNORE_FLAG" == myAlt):
            newSentence = wholeSentence[keyToChange+1:nextKey];
            generatedSentences.append(newSentence)
            continue
        else:
            newSentence = wholeSentence[:]
            newSentence[keyToChange] = myAlt
            generatedSentences.append(newSentence[keyToChange:nextKey]);
        
    return generatedSentences
    
def returnCombinationsOfStrings(sentenceObj):
    
    indexToWordDict = sentenceObj.indexToSetOfWords;
    originalSentence = sentenceObj.ogSentence;
    tokenizedSentence = cleanAndTokenizeText(originalSentence)
    reversedKeys = sorted(indexToWordDict.keys(), reverse=True)
    dictAlternatives  = {}

    sortedKeys = sorted(indexToWordDict.keys())
    sentenceChunks = {}
    htmlChunks = {}
    
    for ind in range(0,len(sortedKeys)):
        key = sortedKeys[ind]
        nextKey = sortedKeys[ind+1] if ind+1 < len(sortedKeys) else len(tokenizedSentence)
        sentenceChunks[key] = generateSentenceChunks(tokenizedSentence, key, nextKey, indexToWordDict[key])

    if(sortedKeys[0] != 0):
        newList = []
        for thislist in sentenceChunks[sortedKeys[0]]:
            newList.append(tokenizedSentence[0:sortedKeys[0]] + thislist)
        sentenceChunks[sortedKeys[0]] = newList;
        
    finalOptions = combineSentenceChunks(tokenizedSentence, sentenceChunks)
    
    finalSentences = []
    for val in finalOptions:
        sentence = ' '.join(val)
        sentiment = senty.polarity_scores(sentence)['compound']
        finalSentences.append(SentenceWithSentiment(sentence,sentiment))
    
    sentenceObj.addFinalSentences(finalSentences)   
    return sentenceObj

In [184]:
def getAlternativeWords(sentenceObj):
    mainSentence = sentenceObj.ogSentence;
    mainSentiment = sentenceObj.ogSentiment;

    sentenceTokens = cleanAndTokenizeText(mainSentence)

    for ind, word in enumerate(sentenceTokens):
        alternativeSentenceWithHTML = []
        copyOfTokens = sentenceTokens[:]
        replacements = []
        replacements = sentenceObj.checkIfWordExists(word)
        if(replacements != []):
            print("FOUND REPEATED WORD {0}".format(word))
            sentenceObj.addAlternativesByIndex(ind, replacements)
            continue;
        
        if(word in LIST_OF_NEGATIONS):
            '''
                If the word is in the list of negations
                We add a [IGNORE_FLAG] tag to the alternatives
                Which will be grabbed in the future by the sentence generation algorithm. (chunking)
            '''
            tmp = ' '.join(sentenceTokens[ind:ind+3])
            if(senty.polarity_scores(tmp)['compound'] != 0):
                sentenceObj.addAlternativesByIndex(ind, ["IGNORE_FLAG", word]);
                continue
        
        score = senty.polarity_scores(word)['compound']
        if(score != 0.0):
            tempReplacements = returnReplacementsForWord(word) # get embedding based relations
            if(VERBOSE_PRINTING): print("Some early replacements for word [{0}] are {1}".format(word, tempReplacements[0:8]))
            if(tempReplacements == []):
                print("No replacements found at all for word {0}".format(word))
                continue
            replacements = posApprovedReplacements(tempReplacements[:], copyOfTokens[:], ind)
            finalReplacements = []
            if(VERBOSE_PRINTING): skippedWords = []
            for new_word in replacements:
                thisSentiment = senty.polarity_scores(new_word)['compound']
                if(thisSentiment != 0.0):
                    finalReplacements.append(new_word)
                else:
                    if(VERBOSE_PRINTING):
                        skippedWords.append(new_word)
            if(VERBOSE_PRINTING and len(skippedWords) > 0):
                print("some skipped words: {0}".format(skippedWords[:5]));
            if(finalReplacements == []):
                if(VERBOSE_PRINTING): print(" -- No POS approved words! -- for word {0}\n some non-POS:{1}".format(word, tempReplacements[:4]))
                continue
            sentenceObj.addAlternativesByIndex(ind, finalReplacements)
            sentenceObj.addWordToAlternatives(word, finalReplacements)
    return sentenceObj




In [185]:
def interactionHelper(dictOfAltObjs, sentiments, thisSent, lastWorkingIndex=0):
    if(thisSent in sentiments):
        mainIndex = sentiments.index(thisSent)
        
    else:
        mainIndex = lastWorkingIndex
    
    numLower =  mainIndex  # Mod to keep it in range 0 -> val 
    numHigher = len(sentiments) - mainIndex - 1

    print("There are totally {0} possible sentences".format(len(sentiments)))
    print("Current sentiment is at index {0}, and length of sentiments is {1}".format(mainIndex, len(sentiments)));
    print(" <--- {0} lower and {1} ---> higher ".format(numLower, numHigher))
    print("Would you like to see a lower sentiment or higher sentiment or ALL?")
    print("Please enter 1 for lower, 2 for higher or 0 for all")
    
    targetSentiment = thisSent
    textInput = input()
    if(textInput == ""):
        print("You entered nothing! exiting")
        return
        
    if(textInput.isdigit()):
        textInput = int(textInput)
        if(textInput == 0):
            if(VERBOSE_PRINTING): print("Printing sentiments from : {0}".format(sentiments))
            for sentiment in sentiments:
                obj = dictOfAltObjs[sentiment][0];
                displayText("{0}  : {1}".format(obj.getHTML(), sentiment))
            return
        elif(textInput == 1 and numLower > 0 and (mainIndex - 1) >= 0):
            targetSentiment = sentiments[mainIndex - 1]
            htmlSentence = dictOfAltObjs[targetSentiment][0].getHTML();
            displayText("{0} : {1}".format(htmlSentence, targetSentiment))
            
        elif(textInput == 2 and numHigher < len(sentiments) and (mainIndex + 1) < len(sentiments)):
            targetSentiment = sentiments[mainIndex + 1]
            htmlSentence = dictOfAltObjs[targetSentiment][0].getHTML();
            displayText("{0} : {1}".format(htmlSentence, targetSentiment))
            
        else:
            print("You entered an invalid option")
            interactionHelper(dictOfAltObjs, sentiments, thisSent, mainIndex)
            return
        
    interactionHelper(dictOfAltObjs, sentiments, targetSentiment,mainIndex)



def interactionWithUser(mainSentenceObj, dictOfAltObjs,sentiments, isUser=False):
    if(isUser == False):
        return
    else:
        mainSentence = mainSentenceObj.ogSentence;
        mainSent = mainSentenceObj.ogSentiment;
        if(VERBOSE_PRINTING):
            print("The following are the sentiment values for the input")
            for ind,sent in enumerate(sentiments):
                print("{0} - {1}".format(ind+1,sent))
        
        interactionHelper(dictOfAltObjs, sentiments, mainSent) 

In [186]:
def extractTwitterDataset():
    df_tweets = pd.read_csv( pathToDatasets + 'cleanedTweets.csv', nrows=NUM_OF_TWEETS, skiprows =TWEET_START)
    tweets = df_tweets.values
    return tweets;

def createDictionaryOfSentStrings(sentencesWithSentiment):
    dict_sentimentToStringObjects = {}
    for obj in sentencesWithSentiment:
        sent = obj.getSentiment()
        if(dict_sentimentToStringObjects.get(sent) == None):
            dict_sentimentToStringObjects[sent] = [obj]
        else:
            (dict_sentimentToStringObjects[sent]).append(obj)

    return dict_sentimentToStringObjects;

def runThroughTweets():
    tweets = extractTwitterDataset()
    
    counterOfTweets = 0;
    for counter,tweet in enumerate(tweets):
        counterOfTweets +=1;
        tweet = tweet[0]
        tweetTokens = cleanAndTokenizeText(tweet)
        mainSentiment = senty.polarity_scores(tweet)['compound']
        if(mainSentiment == 0):
            continue
        print("\n\n Tweet: {0}:{1}\n".format(tweet,mainSentiment))   
        sentenceObj = Sentence(tweet, mainSentiment)
        sentenceObj = getAlternativeWords(sentenceObj)
        replacementDictionary = sentenceObj.getDictOfIndexWords();
        if(len(replacementDictionary) <= 0):        
            print(" -- No new Strings generated ---\n\n")
            continue

        keysToChange = replacementDictionary.keys();
        if(VERBOSE_PRINTING):
            for key in keysToChange:
                print("{0}'th word's ({2}) options: {1}".format(key,replacementDictionary[key],tweetTokens[key]))

        sentenceObj = returnCombinationsOfStrings(sentenceObj)
        sentencesWithSentiment, sentenceObj = printStrings(sentenceObj)
       
        dict_sentimentToStringObjects = createDictionaryOfSentStrings(sentencesWithSentiment)
        
        sentiments = sorted(dict_sentimentToStringObjects.keys())
        
    return counterOfTweets;
        
numOfTweets = runThroughTweets()
print("num of tweets done : {0}".format(numOfTweets))



 Tweet:  that is lame :-0.4215





 Tweet: heroes just is not doing it for me this season... :0.5106





 Tweet: living not downtown sure is not much fun.  :-0.5667





 Tweet:  not calorie wise   i wish junk food was calorie free! i ate a thing of sour skittles and a big ass cherry coke!:-0.6804



--- More options (total: 1575) possible, but not printed ---




 Tweet: man work is hard :-0.1027





 Tweet: getting sick  time for some hot tea studying and then sleeeep:-0.5106





 Tweet: getting eyebrows waxed. more pain :-0.5563





 Tweet: no phantasy star yesterday  going to work...:-0.296





 Tweet: oh  just got all my macheist 3.0 apps  sweet. did not get the espresso serial no though although they said they sent it  oh well :0.4404





 Tweet: picked mich st to win it all from the get go.  was feeling pretty good about that pick all the way up until......tonight.  a is lost too :0.8442



--- More options (total: 5760) possible, but not printed ---




 Tweet: is alone downstairs...working :-0.25





 Tweet: i feel bad for doing it :-0.5423





 Tweet:  is it just me or she hates anoop. i mean seriously she is kinda mean to him. :-0.5574





 Tweet:  yes sprint has 4g only in baltimore and chicago so far :0.4019





 Tweet: i am stuck awake in the middle of the night for the second day in a row. and i felt terrible yesterday. :-0.6249



--- More options (total: 54) possible, but not printed ---




 Tweet: thanks for bursting my bubble :0.4404





 Tweet: going to school soon... can not find anything to wear!! gosh it is so hard :-0.3507





 Tweet:  i was serious  lol:0.3612





 Tweet:  i had on my page for sooooo long!!! until it got deleted  sad day in history:-0.6093





 Tweet: crazy wind today  no birding  :-0.5574





 Tweet: grrr.. my ipods acting weird too! jai ho and thinking of you are not playing the full songs.. ughh. :-0.3786





 Tweet:  send me the dvd cos i have missed out on heaps  not happy about that!:-0.6694





 Tweet:  i am so sorry you are having to go through this. again.  therapyfail:-0.1513





 Tweet: i am not still up i swear. why do i keep losing. gaining. losing. gaining. tweeps  so heart wrenching.:0.1401

FOUND REPEATED WORD losing
FOUND REPEATED WORD gaining




 Tweet: today i realized i am too good at hiding things even i can not find it :0.1779





 Tweet: staying at a friends house...house sitting neighbors are so loudhaving a party :0.7251





 Tweet:  danny im upset that i wasnt here to watch the live chat  i was in a car. for 3 hours on a trip im soooo upset!!:-0.6988

FOUND REPEATED WORD upset


--- More options (total: 256) possible, but not printed ---




 Tweet: downloading nin is new album quotthe slipquot  when the hell did this come out i am so behind the times these days :-0.6808





 Tweet: is watching the hill . . .and its making me sad :-0.4767





 Tweet: so many channels.... yet so so boring... lazy day again  may have to find a hobby:-0.4664





 Tweet:  i miss my buddy  ill be in ny on the 25th:-0.5267





 Tweet:   love the french. i tell people here in the south i am qtr. french and they snarl at me.  french are beautiful people:0.8442





 Tweet: opps.. as i said.. i still got one day remain and now problem come :-0.4019





 Tweet: i activated my selfcontrol block early meaning i can not check out the new qc.  regularizing my internal clock is might be difficult. fb:-0.6597





 Tweet:   oh no:-0.296





 Tweet: spencer is not a good guy. :-0.3412





 Tweet:  what about reese dying on ttsc and season finale next week.  24 boring madame president is a crazy woman.:-0.5719



--- More options (total: 90) possible, but not printed ---




 Tweet:  i hate the limited letterstoo.hope you and the guys are finei pray for my dogshe�s not well :-0.6266



--- More options (total: 700) possible, but not printed ---




 Tweet: did not get shit done today  i am so screwed :-0.2942





 Tweet: too worried and tired to post tonight :-0.6249



--- More options (total: 63) possible, but not printed ---




 Tweet: could not get shit done today  i am so screwed :-0.2942



num of tweets done : 50


## Testing Code


In [187]:
specificWord = "good"
def testOneWord(word=""):
    if(word==""):
        return
    print(word)

In [188]:
specificString = ""
def specificString(textString=""):
    if(textString == "" or textString == None):
        return
    textString = dataClean(textString)
    textTokens = cleanAndTokenizeText(textString)
    mainSentiment = senty.polarity_scores(textString)['compound']
    if(mainSentiment == 0):
        print("{} \n No sentiment found in sentence".format(textString));
        return;
    print("\n {0}:{1}\n".format(textString,mainSentiment))   
    sentenceObj = Sentence(textString, mainSentiment)
    sentenceObj = getAlternativeWords(sentenceObj)
    replacementDictionary = sentenceObj.getDictOfIndexWords();
    if(len(replacementDictionary) <= 0):        
        print(" -- No new Strings generated ---\n\n")
        return
    
    keysToChange = replacementDictionary.keys();
    if(VERBOSE_PRINTING):
        for key in keysToChange:
               print("{0}'th word's ({2}) options: {1}".format(key,replacementDictionary[key],textTokens[key]))
    
    sentenceObj = returnCombinationsOfStrings(sentenceObj)
    allPossibleSentences, sentenceObj = printStrings(sentenceObj)
    
    
    dict_sentimentToStringObjects = createDictionaryOfSentStrings(allPossibleSentences)
        
    sentiments = sorted(dict_sentimentToStringObjects.keys())

print("Enter a string to use")


inputText = ""
# inputText = input()
if(inputText == ""):
    return
elif(inputText == "t"):
#     specificString("I really love hot chocolate, but I'm not good with hot coffee")
#     specificString("damn! the grind is inspirational and saddening at the same time. do not want you to stop cuz i like what u do! much love");
    specificString("the grind is inspirational and saddening at the same time. do not want you to stop cuz i like what u do!")
else:
    specificString(inputText)
# I really love hot chocolate, but I'm not good with hot coffee 

Enter a string to use




this is making me feel bad:-0.5423
this is making me feel a bit bad:-0.5423

this is making me feel very bad:-0.5849
this is making me feel quite bad:-0.5849
this is making me feel incredibly bad:-0.5849

this is making me feel kinda bad:-0.4951
this is making me feel slightly bad:-0.4951
this is making me feel barely bad:-0.4951

this is making me feel not bad:0.431
